content/articles/2018-02-usethis-1-3-0.Rmd

```diff
@@ -0,0 +1,106 @@
+---
+title: usethis 1.3.0
+slug: usethis-1-3-0
+description: >
+  usethis 1.3.0 is now on CRAN.
+author: Jenny Bryan
+date: 2018-02-23
+categories:
+  - package
+tags: []
+photo:
+  url: https://pixabay.com/en/tools-vintage-woodworking-saw-1209764/
+  author: unknown
+---
+
+```{r setup, include = FALSE}
+library(usethis)
+knitr::opts_chunk$set(collapse = TRUE, comment = "#>")
+```
+
+We are pleased to announce that [usethis](http://usethis.r-lib.org) 1.3.0 is now available on CRAN. usethis automates many common setup tasks for R projects, especially, but not only, for R packages. You can learn more about it at <http://usethis.r-lib.org>. Install it with:
+
+```{r, eval = FALSE}
+install.packages("usethis")
+```
+
+usethis is part of the refactoring of the [devtools](https://github.com/r-lib/devtools#readme) package and was [first released in late 2017](https://www.tidyverse.org/articles/2017/11/usethis-1.0.0/), at version 1.0.0. We say a bit more about its past and future below and then highlight notable developments in this version. For the full details, see the [release notes](https://github.com/r-lib/usethis/releases/tag/v1.3.0).
+
+## usethis vs. devtools
+
+[devtools](https://github.com/r-lib/devtools#readme) is an R package that helps you create, develop, and install R packages. By 2017, it was clear that devtools would be easier to maintain if it was split into a few smaller and more focused packages. usethis is the first of these to be released publicly on CRAN. usethis is the home for functions related to package and project setup.
+
+devtools will continue to exist as a meta-package, exposing functionality from lower-level, domain-specific packages. In the long-run, a useR should attach devtools, via `library("devtools")`, and then call devtools functions in their daily workflow, as usual. There are two reasons you might need to know about our internal refactoring:
+
+  * The Awkward Adolescent Phase. It is impossible to choreograph the release of a suite of packages on CRAN. Therefore, there may be periods when functionality exists in, e.g., usethis, that is not yet exposed by devtools. During such times, you would access those specific functions directly through usethis. This implies you should attach devtools first and then any of its constituent packages, as they roll out. This applies to the interactive use of usethis/devtools functions in your daily workflow.
+  * Programming and Development. If you are using devtools functions programmatically, i.e. in production scripts or in another package, it is recommended to do so via the function's primary package home and not through devtools.
```

Comment:
I wonder if we should be stronger about this (remove 'it is recommend to') and give some justification about the reasons.

1. lighter dependencies for your package (you only need features you use)
2. easier to release new versions of devtools (fewer reverse dependencies)
3. finer grained versioning (can depend on pkgbuild 1.2.3 and pkgload 4.5.6 even if devtools uses pkgbuild 4.5.6 and pkgload 4.5.6)

Reply:
Add an example here, perhaps?

Reply:
And maybe an analogy to tidyverse (and reinforce that's it's bad to this for tidyverse too)

Maybe `use_package()` should warn if you try and add a dependency to devtools?